isis/src/base/apps/framestitch/framestitch.cpp

```diff
@@ -0,0 +1,220 @@
+
+    // It is very important that the even cube gets added first as later on
+    // we'll use frameNumber % 2 to index the input cube vector
+    QString evenCubeFile = ui.GetCubeName("EVEN");
+    Cube* evenCube = process.SetInputCube(evenCubeFile,
+                                          CubeAttributeInput(evenCubeFile));
+    QString oddCubeFile = ui.GetCubeName("ODD");
+    Cube* oddCube = process.SetInputCube(oddCubeFile,
+                                         CubeAttributeInput(oddCubeFile));
+
+
+    // Check that all the inputs are valid
+    if (evenCube->sampleCount() != oddCube->sampleCount() ||
+        evenCube->lineCount() != oddCube->lineCount() ||
+        evenCube->bandCount() != oddCube->bandCount()) {
+      QString msg = "Even and odd cube dimensions must match.";
+      throw IException(IException::User, msg, _FILEINFO_);
+    }
+
+    bool inputFlipped = false;
+
+    if (evenCube->hasGroup("Instrument") && oddCube->hasGroup("Instrument")) {
+      const PvlGroup &evenInst = evenCube->group("Instrument");
+      const PvlGroup &oddInst = oddCube->group("Instrument");
+      // Use the start time as an indicator of being the same original image
+      if (QString::compare(QString(evenInst["StartTime"]), QString(oddInst["StartTime"])) != 0) {
+        QString msg = "Even and odd cube start times must match.";
+        throw IException(IException::User, msg, _FILEINFO_);
+      }
+
+      if (evenInst.hasKeyword("DataFlipped") && oddInst.hasKeyword("DataFlipped")) {
+        if (toBool(evenInst["DataFlipped"]) != toBool(oddInst["DataFlipped"])) {
+          QString msg = "Both input cubes must be flipped or not flipped. Cannot combine "
+                        "a flipped and unflipped cube.";
+          throw IException(IException::User, msg, _FILEINFO_);
+        }
+        inputFlipped = toBool(evenInst["DataFlipped"]);
+      }
+    }
+
+    QString outCubeFile = ui.GetCubeName("TO");
+    Cube *outCube = process.SetOutputCube(
+          outCubeFile, CubeAttributeOutput(outCubeFile),
+          evenCube->sampleCount(), evenCube->lineCount(), evenCube->bandCount());
+
+    int frameSize = 1;
+    if (ui.WasEntered("FRAMESIZE")) {
+      frameSize = ui.GetInteger("FRAMESIZE");
+    }
+    else {
+      // User didn't pass the size of the frames so attempt to infer it from
+      // the cubes
+      int evenFrameSize = computeFrameSize(evenCube);
+      int oddFrameSize = computeFrameSize(oddCube);
+
+      if (evenFrameSize == 0) {
+        QString msg = "Failed to find any NULL frames in even cube when attempting "
+                      "to compute frame size.";
+        throw IException(IException::User, msg, _FILEINFO_);
+      }
+      if (oddFrameSize == 0) {
+        QString msg = "Failed to find any NULL frames in odd cube when attempting "
+                      "to compute frame size.";
+        throw IException(IException::User, msg, _FILEINFO_);
+      }
+      if (evenFrameSize != oddFrameSize) {
```

Comment:
This implicitly assumes that all frames in both even and odd cubes have the same size. If they don't, given that the frame size is found using a median, these two medians will likely disagree.

Reply:
Then finding the median is not helpful as all values should be equal. Instead, when, for a given even or odd cube the height of each frame is found, it should be verified right there that all obtained frames in a cube have the same height, before comparing the frame heights between the cubes.

Reply:
I went with the median to account for any weird situations where there's random lines of NULL in the image that are not associated with missing even/odd frames. I could see just validating here being a better solution though.

Reply:
Sounds good. That was more for my own understanding.